### PR TITLE
Fix php ini parsing from the CLI

### PR DESCRIPTION
**Pull request by dingo-d**
The rebasing didn't work as I thought it would on the previous PR so I just cherry-picked the commits against a correct branch.

#### Original message

This is a (possible) fix for the PHPUnit issue https://github.com/sebastianbergmann/phpunit/issues/4664

I've added the suggestion by @krakjoe ([link](https://github.com/sebastianbergmann/phpunit/issues/4664#issuecomment-835353679)) and some tests to cover this part of the code.

I've run the tests on the PHPUnit locally on the latest branch to check if anything broke, and the existing tests passed. I guess this should be tested a bit thoroughly.

When I ran the tests on the PHPUnit repo and put the breakpoint to inspect the result of merging the settings with `getCurrentSettings` method in the `AbstractPhpProcess.php` file inside the `getCommand` method I could see my xdebug settings correctly parsed, so I think there should be no breaks.

I had to add the updated code manually inside the vendor file to test this because I got a composer error when trying to use my branch instead of the v6 for the `php-code-coverage` package.

```bash
 Problem 1
    - phpunit/php-code-coverage dev-master requires sebastian/environment ^6.0 -> found sebastian/environment[dev-master, 6.0.x-dev (alias of dev-master)] but it conflicts with your root composer.json require (dev-fix-php-ini-parsing).
    - phpunit/php-code-coverage 10.0.x-dev is an alias of phpunit/php-code-coverage dev-master and thus requires it to be installed too.
    - Root composer.json requires phpunit/php-code-coverage ^10.0 -> satisfiable by phpunit/php-code-coverage[10.0.x-dev (alias of dev-master)].
```

This is the image of the breakpoint and part of the `$settings` array:

image

I'd love to have somebody test it a bit more just to be sure.

Oh and I'd love to add a co-authored commit for @krakjoe but I'm not sure what the no-reply email is 🙈 I can update the commit message to add it once I find out the correct email 🙂

### CODE REVIEW

File: src/Runtime.php [hunk 1]
```diff
@@ -289,8 +289,12 @@ public function getCurrentSettings(array $values): array
             foreach ($values as $value) {
                 $set = ini_get($value);
 
-                if (isset($config[$value]) && $set != $config[$value]) {
```
Comment (sebastianbergmann) [inline src/Runtime.php:292]: Can we use type-safe comparison here?

File: src/Runtime.php [hunk 2]
```diff
@@ -289,8 +289,12 @@ public function getCurrentSettings(array $values): array
...
-                    $diff[] = sprintf('%s=%s', $value, $set);
+                if ($set == null) {
```
Comment (sebastianbergmann) [inline src/Runtime.php:292]: Can we use type-safe comparison here?

File: src/Runtime.php [hunk 3]
```diff
@@ -289,8 +289,12 @@ public function getCurrentSettings(array $values): array
...
+                    continue;
+                }
+
+                if ((!isset($config[$value]) || ($set != $config[$value]))) {
```
Comment (sebastianbergmann) [inline src/Runtime.php:296]: Can we use type-safe comparison here?

File: tests/RuntimeTest.php [hunk 4]
```diff
@@ -142,6 +142,30 @@ public function testGetVendorUrlReturnsPhpDotNetWhenRunningPhp(): void
         $this->assertSame('https://secure.php.net/', $this->env->getVendorUrl());
     }
 
+    public function testDiscardingCommentsReturnsFalseIfOpcacheIsNotActive(): void
```
Comment (sebastianbergmann) [inline tests/RuntimeTest.php:145]: This test looks unrelated to the topic of the PR. What am I missing?

Comment (dingo-d) [reply tests/RuntimeTest.php:145]: Oh, I noticed that this part was uncovered in the coverage so I just added it. I can remove it so that the tests are related only to the above change 👍🏼

Comment (sebastianbergmann) [reply tests/RuntimeTest.php:145]: Yes, please. As it is, this test might fail depending on the environment (pun intended).
